publish/publish.sh

```diff
@@ -0,0 +1,179 @@
+#!/bin/bash
+
+# A script to publish the smart contract under ../move/<dir name>
+# On successful deployment, the script updates the env files publish/.env and app/.env
+# The env files are updated with env variables to represent the newly created objects
+
+# dir of smart contract
+# TODO: update dir name as needed
+MOVE_PACKAGE_PATH="../move"
+
+
+# check this is being run from the right location
+if [[ "$PWD" != *"/publish" ]]; then
+    echo "Please run this from ./publish"
+    exit 0
+fi
+
+# check dependencies are available
+for dep in jq curl sui; do
+    if !command -V ${i} 2>/dev/null; then
+        echo "Please install lib ${dep}"
+        exit 1
+    fi
+done
+
+# process command line args
+ENV=
+DRY_RUN=false
+
+while [ $# -gt 0 ]; do
+    case "$1" in
+    --env=*)
+        ENV="${1#*=}"
+        case ${ENV} in
+        "testnet") ;;
+        "devnet") ;;
+        "local") ;;
+        *) echo "unknown env {$ENV}"
+            exit 1
+        esac #end inner case
+        ;;
+    --dry-run)
+        DRY_RUN=true
+        ;;
+    *)
+        echo "Unknown arg $1"
+        exit 1
+    esac #end case
+    shift
+done
+
+if [ -z "${ENV}" ]; then
+    echo "Usage: ./publish.sh --env={testnet|devnet|local} [--dry-run]"
+    echo "Dry run will not publish contract"
+    exit 1
+fi
+
+# check for admin address env var
+if [ -z "${ADMIN_ADDRESS}" ]; then
+    echo "Please setup env var ADMIN_ADDRESS with the client address to publish from"
+    echo "E.g. > export ADMIN_ADDRESS=\$(sui client active-address)"
+    exit 1
+fi
+
+# check for admin secret key
+if [ -z "${ADMIN_SECRET_KEY}" ]; then
```

Comment:
ideally, we make this optional.
if the private key already exists on the host, we don't need to provide it, `sui client` should be able to handle this